kvdb/sqlbase/db.go

```diff
@@ -345,7 +345,7 @@ func (db *db) executeTransaction(f func(tx walletdb.ReadWriteTx) error,
 			}
 		}
 
-		return commitErr
+		return nil
```

Comment:
I think this should still return the `commitErr` as otherwise, if it can't be mapped to a `SQLError`, then we always return `nil`, when the actual error should be thread through.

Reply:
Also revisiting this section, it should be trying to map the `commitErr` here, now `fnErrr` (shadowed from above).